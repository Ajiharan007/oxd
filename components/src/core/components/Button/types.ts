export const SIZE_SMALL = 'small';
export const SIZE_MEDIUM = 'medium';
export const SIZE_LARGE = 'large';

export type ButtonSize =
  | typeof SIZE_SMALL
  | typeof SIZE_MEDIUM
  | typeof SIZE_LARGE;
