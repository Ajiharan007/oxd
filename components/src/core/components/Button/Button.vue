<template>
  <button type="button" :class="classes" @click="onClick" :style="style">
    {{ label }}
  </button>
</template>

<script>
export default {
  name: "oxd-button",

  props: {
    label: {
      type: String,
      required: true,
    },
    primary: {
      type: Boolean,
      default: false,
    },
    size: {
      type: String,
      default: "medium",
      validator: function(value) {
        return ["small", "medium", "large"].indexOf(value) !== -1;
      },
    },
    backgroundColor: {
      type: String,
    },
  },

  computed: {
    classes() {
      return {
        "oxd-button": true,
        "oxd-button--primary": this.primary,
        "oxd-button--secondary": !this.primary,
        [`oxd-button--${this.size}`]: true,
      };
    },
    style() {
      return {
        backgroundColor: this.backgroundColor,
      };
    },
  },

  methods: {
    onClick() {
      this.$emit("onClick");
    },
  },
};
</script>

<style lang="scss" scoped>
@import "../../../styles";

.oxd-button {
  font-weight: 600;
  border: 0;
  border-radius: 3em;
  cursor: pointer;
  display: inline-block;
  line-height: 1;
}
.oxd-button--primary {
  color: $oxd-white-color;
  background-color: $oxd-primary-one-color;
}
.oxd-button--secondary {
  color: $oxd-white-color;
  background-color: $oxd-secondary-one-color;
}
.oxd-button--small {
  font-size: 12px;
  padding: 10px 16px;
}
.oxd-button--medium {
  font-size: 14px;
  padding: 11px 20px;
}
.oxd-button--large {
  font-size: 16px;
  padding: 12px 24px;
}
</style>
