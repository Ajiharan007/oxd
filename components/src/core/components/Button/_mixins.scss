@mixin oxd-button-color($background-color) {
  color: $oxd-button-font-color;
  background-color: $background-color;
}

@mixin oxd-button-color-ghost($color) {
  color: $color;
  border: $oxd-button-border--ghost;
  background-color: transparent;
}
