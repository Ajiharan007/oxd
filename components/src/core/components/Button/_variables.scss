$oxd-button-min-width: 100px !default;
$oxd-button-horizontal-padding: 32px !default;
$oxd-button-vertical-padding--large: 16px !default;
$oxd-button-vertical-padding--medium: 8px !default;
$oxd-button-vertical-padding--small: 4px !default;
$oxd-button-font-size--large: 14px !default;
$oxd-button-font-size--medium: 14px !default;
$oxd-button-font-size--small: 12px !default;
$oxd-button-font-weight: 600 !default;
$oxd-button-font-family: $oxd-font-family !default;
$oxd-button-font-color: $oxd-white-color !default;

$oxd-button-color--main: $oxd-primary-one-color !default;
$oxd-button-color--main-pressed: #ff6c03 !default;
$oxd-button-color--main-hover: #ff8a36 !default;
$oxd-button-color--main-inactive: $oxd-interface-gray-lighten-2-color !default;

$oxd-button-color--secondary: $oxd-secondary-four-color !default;
$oxd-button-color--secondary-pressed: #68a61d !default;
$oxd-button-color--secondary-hover: #84d225 !default;
$oxd-button-color--secondary-inactive: $oxd-interface-gray-lighten-2-color !default;

$oxd-button-color--ghost: $oxd-secondary-four-color !default;
$oxd-button-border--ghost: $oxd-border !default;
$oxd-button-color--ghost-pressed: #68a61d !default;
$oxd-button-color--ghost-hover: #84d225 !default;
$oxd-button-color--ghost-inactive: $oxd-interface-gray-lighten-1-color !default;
