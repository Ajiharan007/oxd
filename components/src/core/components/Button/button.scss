@import '../../../styles';
@import 'variables';
@import 'mixins';

.oxd-button {
  font-family: $oxd-button-font-family;
  font-weight: $oxd-button-font-weight;
  border: 0;
  border-radius: 3em;
  cursor: pointer;
  display: inline-block;
  line-height: 1;
  padding-right: $oxd-button-horizontal-padding;
  padding-left: $oxd-button-horizontal-padding;
  min-width: $oxd-button-min-width;
}

/* Main Button */
.oxd-button--main {
  @include oxd-button-color($oxd-button-color--main);
}

.oxd-button--main:hover {
  @include oxd-button-color($oxd-button-color--main-hover);
}

.oxd-button--main:active {
  @include oxd-button-color($oxd-button-color--main-pressed);
}

.oxd-button--main:disabled {
  @include oxd-button-color($oxd-button-color--main-inactive);
}

/* Secondary Button */
.oxd-button--secondary {
  @include oxd-button-color($oxd-button-color--secondary);
}

.oxd-button--secondary:hover {
  @include oxd-button-color($oxd-button-color--secondary-hover);
}

.oxd-button--secondary:active {
  @include oxd-button-color($oxd-button-color--secondary-pressed);
}

.oxd-button--secondary:disabled {
  @include oxd-button-color($oxd-button-color--secondary-inactive);
}

/* Ghost Button */
.oxd-button--ghost {
  @include oxd-button-color-ghost($oxd-button-color--ghost);
}

.oxd-button--ghost:hover {
  @include oxd-button-color-ghost($oxd-button-color--ghost-hover);
}

.oxd-button--ghost:active {
  @include oxd-button-color-ghost($oxd-button-color--ghost-pressed);
}

.oxd-button--ghost:disabled {
  @include oxd-button-color-ghost($oxd-button-color--ghost-inactive);
}

/* Button Sizes */
.oxd-button--small {
  font-size: $oxd-button-font-size--small;
  padding-top: $oxd-button-vertical-padding--small;
  padding-bottom: $oxd-button-vertical-padding--small;
}

.oxd-button--medium {
  font-size: $oxd-button-font-size--medium;
  padding-top: $oxd-button-vertical-padding--medium;
  padding-bottom: $oxd-button-vertical-padding--medium;
}

.oxd-button--large {
  font-size: $oxd-button-font-size--large;
  padding-top: $oxd-button-vertical-padding--large;
  padding-bottom: $oxd-button-vertical-padding--large;
}
