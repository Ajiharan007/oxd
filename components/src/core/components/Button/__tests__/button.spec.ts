import {shallowMount} from '@vue/test-utils';
import Button from '@orangehrm/oxd/core/components/Button/Button.vue';

describe('Button.vue', () => {
  it('renders OXD button', () => {
    const label = 'Button';
    const wrapper = shallowMount(Button, {
      props: {label},
    });
    expect(wrapper.text()).toMatch(label);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
