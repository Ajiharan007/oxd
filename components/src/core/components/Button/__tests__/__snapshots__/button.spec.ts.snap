// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue renders OXD button 1`] = `<button type="button" class="oxd-button oxd-button--medium oxd-button--main">Button</button>`;

exports[`Button.vue should renders OXD button with custom color 1`] = `<button type="button" class="oxd-button oxd-button--medium oxd-button--main" style="background-color: palegreen;">Button</button>`;

exports[`Button.vue should renders OXD ghost button 1`] = `<button type="button" class="oxd-button oxd-button--medium oxd-button--ghost">Button</button>`;

exports[`Button.vue should renders OXD large button 1`] = `<button type="button" class="oxd-button oxd-button--large oxd-button--main">Button</button>`;

exports[`Button.vue should renders OXD main button 1`] = `<button type="button" class="oxd-button oxd-button--medium oxd-button--main">Button</button>`;

exports[`Button.vue should renders OXD medium button 1`] = `<button type="button" class="oxd-button oxd-button--medium oxd-button--main">Button</button>`;

exports[`Button.vue should renders OXD secondary button 1`] = `<button type="button" class="oxd-button oxd-button--medium oxd-button--secondary">Button</button>`;

exports[`Button.vue should renders OXD small button 1`] = `<button type="button" class="oxd-button oxd-button--small oxd-button--main">Button</button>`;
